db/migrate/20160919080104_add_foreign_keys.rb

```diff
@@ -0,0 +1,18 @@
+class AddForeignKeys < ActiveRecord::Migration
+  def change
+    reversible do |dir|
+      dir.up do
+        Alchemy::Cell.unscoped.all.select { |c| c.page.nil? && c.page_id.present? }.each(&:destroy)
+        Alchemy::Content.unscoped.all.select { |c| c.essence.nil? && c.essence_id.present? }.each(&:destroy)
+        Alchemy::Content.unscoped.all.select { |c| c.element.nil? && c.element_id.present? }.each(&:destroy)
+        Alchemy::Element.unscoped.all.select { |e| e.page.nil? && e.page_id.present? }.each(&:destroy)
+        Alchemy::Element.unscoped.all.select { |e| e.cell.nil? && e.cell_id.present? }.each(&:destroy)
+      end
+    end
+
+    add_foreign_key :alchemy_cells, :alchemy_pages, column: :page_id, on_update: :cascade, on_delete: :cascade, name: :alchemy_cells_page_id_fkey
```

Comment:
I always add the update, because if you change the id of the page (for whatever reason), the db should keep the link, otherwise the link would be broken.